rule vcf_list:
        """
        Create the txt file containing the list of the VCF files to merge
        """
        input:
		"{wdir}/{sra}_cutesv.vcf"
        output:
                "{wdir}/{sra}_vcf_list.txt"
        shell:
		"""
		echo '{wdir}/{sra}_svim.vcf' > {wdir}/{sra}_vcf_list.txt
		echo '{wdir}/{sra}_sniffles.vcf' >> {wdir}/{sra}_vcf_list.txt
		echo '{wdir}/{sra}_cutesv.vcf' >> {wdir}/{sra}_vcf_list.txt
		"""

rule jasmine:
        """
        Merge the VCF files obtained by the three SV callers
        """
        input:
                "{wdir}/{sra}_vcf_list.txt"
        output:
                "{wdir}/{sra}_merged.vcf"
        conda:
                "Envs/jasminesv.yaml"
        shell:
                "Scripts/jasmine.sh {sra}"

rule svjedi_graph:
        """
        Genotype the SVs
        """
        input:
                "{wdir}/{sra}_merged.vcf"
        output:
                "{wdir}/{sra}_genotyped.vcf"
        conda:
                "Envs/svjedi.yaml"
        shell:
                "Scripts/svjedigraph.sh"


rule nanoplot:
	"""
	Quality control of raw data
	"""
	input:
		"{wdir}/{sra}_fastq.gz"
	output:
		"{wdir}/{sra}_nanoplotstats.txt
	conda:
		"Envs/nanoplot.yaml"
	shell:
		"Scripts/nanoplot.sh {sra}"

rule samtools_stats:
	"""
	Quality control of the mapping
	"""
	input:
		"{wdir}/{sra}_sorted.bam"
	output:
		"{wdir}/{sra}_mapping.stats"
	conda:
		"Envs/samtools.yaml"
	shell:
		"Scripts/samtools_stats.sh {sra}"

rule bamstats:
	"""
	Vizualisation of quality control of the mapping
	"""
	input:
		"{wdir}/{sra}_mapping.stats"
	output:
		"{wdir}/{sra}_mapping_visu"
	conda:
		"Envs/gnuplot.yaml"
	shell:
		"Scripts/bamstats.sh {sra}"


