# Snakemake pipeline for detection and analysis of structural variants in reference genomes

This pipeline detects heterozygous SVs using three tools: SVIM, Sniffles and CuteSV.

## Installation

The following dependencies must be installed before launching the pipeline :
* conda
* snakemake

## Usage

You can launch the pipeline in two ways: either by running the *launcher.sh* script, or by executing the following command in the terminal :
```
snakemake -s pipeline.snake --config config.yaml --use-conda --conda-frontend "mamba" --cores 8 --rerun-incomplete
```

## Input data

The input data are the accession numbers of the long-reads (ERR) and the reference genome (GCA).

## Output files

The output file is a VCF file containing the list of SVs, named *ERRxxx_merged.vcf*. It is the result of merging the three SV catalogues generated by the SVIM, Sniffles and CuteSV tools.
